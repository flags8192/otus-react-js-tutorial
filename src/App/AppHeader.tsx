import React, { FC, memo, useContext } from 'react';
import Panel from '@src/layout/Panel';
import styled from '@emotion/styled';
import Button from '@src/components/Button';
import store from '@src/store';
import { resetApp, setLevel } from '@src/App/actions';
import Select from '@src/components/Select';

const Title = styled.p`
  margin: auto 20px;
`;

const AppHeader: FC = () => {
  const { dispatch } = useContext(store);

  const handleReset = () => {
    dispatch(resetApp());
  };

  const handleSelectLevel = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const { value } = event.currentTarget;
    dispatch(setLevel(value));
  };

  return (
    <Panel role="topPanel">
      <Title>Lines</Title>
      <Button onClick={handleReset} title="Reset" />
      <Select onSelect={handleSelectLevel} />
    </Panel>
  );
};

export default memo(AppHeader);
