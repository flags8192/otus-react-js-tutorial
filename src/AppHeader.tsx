import React, { PureComponent } from 'react';
import styled from '@emotion/styled';
import Input from '@src/common/Input';

const Root = styled.div`
  height: 20vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

const Title = styled.div`
  margin: 10px;
`;

interface IAppHeader {
  onChangeFilter: (val: string) => void;
}
export class AppHeader extends PureComponent<IAppHeader> {
  constructor(props: IAppHeader) {
    super(props);
  }

  render() {
    return (
      <Root>
        <Title role="headerTitle">Дз №4</Title>
        <Input onChange={this.props.onChangeFilter} autoFocus={true} />
      </Root>
    );
  }
}
