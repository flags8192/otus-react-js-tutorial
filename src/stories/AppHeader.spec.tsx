import React from 'react';
import { render, screen } from '@testing-library/react';
import { AppHeader } from '@src/AppHeader';

describe('AppHeader', () => {
  it('Render <AppHeader>', () => {
    const handleChangeFilter = jest.fn();
    const { asFragment } = render(<AppHeader onChangeFilter={handleChangeFilter} />);
    expect(asFragment()).toMatchSnapshot();
    const headerTitle = screen.getByRole(/headerTitle/gi);
    expect(headerTitle).toBeInTheDocument();
    const filterInput = screen.getByRole(/filterInput/gi);
    expect(filterInput).toBeInTheDocument();
  });
});
