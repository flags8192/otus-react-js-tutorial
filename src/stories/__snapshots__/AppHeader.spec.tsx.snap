// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader Render <AppHeader> 1`] = `
<DocumentFragment>
  <div
    class="css-x0l4uv"
  >
    <div
      class="css-1lu5s7y"
      role="headerTitle"
    >
      Дз №4
    </div>
    <div
      class="css-1ng1nrc"
    >
      Поиск:
      <input
        role="filterInput"
      />
    </div>
  </div>
</DocumentFragment>
`;
